research/audioset/yamnet/export.py

```diff
@@ -0,0 +1,208 @@
+"""Exports YAMNet as: TF2 SavedModel, TF-Lite model, TF-JS model.
+
+The exported models all accept as input:
+- 1-d float32 Tensor of arbitrary shape containing an audio waveform
+  (assumed to be mono 16 kHz samples in the [-1, +1] range)
+and return as output:
+- a 2-d float32 Tensor of shape [num_frames, num_classes] containing
+  predicted class scores for each frame of audio extracted from the input.
+- a 2-d float32 Tensor of shape [num_frames, embedding_size] containing
+  embeddings of each frame of audio.
+- a 2-d float32 Tensor of shape [num_spectrogram_frames, num_mel_bins]
+  containing the log mel spectrogram of the entire waveform.
+The SavedModels will also contain (as an asset) a class map CSV file that maps
+class indices to AudioSet class names and Freebase MIDs.
+
+Requires pip-installing tensorflow_hub and tensorflowjs.
+
+Usage:
+  export.py <path/to/YAMNet/weights-hdf-file> <path/to/output/directory>
+and the various exports will be created in subdirectories of the output directory.
+Assumes that it will be run in the yamnet source directory from where it loads
+the class map. Skips an export if the corresponding directory already exists.
+"""
+
+import os
+import sys
+import tempfile
+import time
+
+import numpy as np
+import tensorflow as tf
+assert tf.version.VERSION >= '2.0.0', (
+    'Need at least TF 2.0, you have TF v{}'.format(tf.version.VERSION))
+import tensorflow_hub as tfhub
+from tensorflowjs.converters import tf_saved_model_conversion_v2 as tfjs_saved_model_converter
+
+import params as yamnet_params
+import yamnet
+
+
+def log(msg):
+  print('\n=====\n{} | {}\n=====\n'.format(time.asctime(), msg), flush=True)
+
+
+class YAMNet(tf.Module):
+  "''A TF2 Module wrapper around YAMNet."""
+  def __init__(self, weights_path, params):
+    super().__init__()
+    self._yamnet = yamnet.yamnet_frames_model(params)
+    self._yamnet.load_weights(weights_path)
+    self._class_map_asset = tf.saved_model.Asset('yamnet_class_map.csv')
```

Comment:
Do we ever test/demonstrate using this Asset?  It looks like check_model reads the one in the local directory.

Reply:
Good point, done.